flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStore.java

```diff
@@ -175,28 +181,33 @@ private MergeEngine mergeEngine() {
         return options.get(MERGE_ENGINE);
     }
 
-    private FileStore buildFileStore() {
+    FileStore buildFileStore() {
         RowType partitionType = TypeUtils.project(type, partitionKeysIndex());
         FileStoreOptions fileStoreOptions = new FileStoreOptions(options);
+        String json = options.get(COMPACTION_SCANNED_MANIFEST);
+        PartitionedManifestMeta manifestMeta =
+                json == null ? null : PartitionedManifestMeta.fromJson(json);
```

Comment:
Q: you mean people will need to set a string with json serialized `PartitionedManifestMeta` when creating a managed flink native table ?  If so, I worry that the bar is too high for users..

Reply:
> Q: you mean people will need to set a string with json serialized `PartitionedManifestMeta` when creating a managed flink native table ? If so, I worry that the bar is too high for users..

This will not expose to users; during the planning phase, the JSON string is serialized from the collected manifest entries and injected into the option, and `TableStore` will get the JSON from the option and deserialize it.